nebula-logger/core/main/log-management/classes/LogEntryHandler.cls

```diff
@@ -44,16 +45,22 @@ public without sharing class LogEntryHandler extends LoggerSObjectHandler {
     }
 
     private void setApexClassFields() {
-        List<String> apexClassNames = new List<String>();
-        List<LogEntry__c> apexLogEntries = new List<LogEntry__c>();
+        Set<String> apexClassNames = new Set<String>();
+        List<LogEntry__c> apexClassLogEntries = new List<LogEntry__c>();
         for (LogEntry__c logEntry : this.logEntries) {
-            if (logEntry.OriginType__c == 'Apex' && String.isNotBlank(logEntry.OriginLocation__c)) {
-                // OriginLocation__c contains the class name + method name. Examples:
-                // MyClass.someMethod
-                // MyClass.MyInnerClass.someMethod
-                String apexClassName = logEntry.OriginLocation__c.substringBefore('.');
-                apexClassNames.add(apexClassName);
-                apexLogEntries.add(logEntry);
+            Boolean isApexClassLogEntry = false;
+            if (logEntry.OriginSourceMetadataType__c == 'ApexClass' && String.isNotBlank(logEntry.OriginSourceApiName__c)) {
+                isApexClassLogEntry = true;
+                apexClassNames.add(logEntry.OriginSourceApiName__c);
+            }
+
+            if (logEntry.ExceptionSourceMetadataType__c == 'ApexClass' && String.isNotBlank(logEntry.ExceptionSourceApiName__c)) {
```

Comment:
I can't tell if the above statement and this one are mutually exclusive or not. I realize they're set up to not be that way, but just looking at `isApexClassLogEntry` I can't tell if making `apexClassLogEntries` into a set and always adding in each conditional might be cleaner. What do you think?

Reply:
Yeah, they are not mutually exclusive - you could have 1 `LogEntry__c` that originated in an Apex class, and the logged exception also came from an Apex class. (You could also have other combos, like a `LogEntry__c` that originated in an Apex trigger/Flow/LWC, and a logged exception from an Apex class).

I'll try to use a set, I think you're right that that would be cleaner. I tend to avoid sets for non-primitives, but in this case, adding the same `LogEntry__c` twice to a set should work without any issues.